### PR TITLE
ST: Log collector for failed tests

### PR DESCRIPTION
**Pull request by serrss**
### Type of change
- Logger

### Description
This PR has logic to print the cluster events and logs for each pod in a specific namespace at the moment when the test fails by following reasons:
1. Timeout
2. Assert error
3. Kubernetes client error

Log collector uses the fabric8 client to get all events and logs in a specific namespace if during executing some system test JUnit caught expected exceptions. Expected 
exceptions can be caught by the method `handleTestExecutionException()` in `StrimziExtension` class. 

During this task, I found the following problems for  logging in the current framework:
1. Problem about JUnit listeners. JUnit doesn't have any listener to execute some custom code between test method and after the test method. All we can do is just use listener `TestExecutionListener` and Override `executionFinished` but this means that if a test fails, JUnit will execute `@AfterEach` methods before `executionFinished`, so all resource in OpenShift will be deleted in `executionFinished` phase.
As some solution, we can start writing logs to some files by `executionStarted` and then finish writing by `executionFinished` but it can make some zombie process or get some VM resources.

2. We have 2 classes where we create resources for ST. First place is `StrimziExtension` and second place is `Resource`. And this is a reason that we don't have a vision what we created by `StrimziExtension` and what we created by `Resource`, so we can't implement some logic which will work with 2 places simultaneously but we can use OpenShift API to grep all information that we need.

### Checklist
- [x] Add ability to write logs in file
- [ ] Remove current solution for CO 
- [ ] Link logs to failed test method

### CODE REVIEW

File: api/src/test/java/io/strimzi/test/StrimziExtension.java [hunk 1]
```diff
@@ -92,11 +99,14 @@
     private static final String DEFAULT_TAG = "";
     private static final String TAG_LIST_NAME = "junitTags";
     private static final String START_TIME = "start time";
+    private static final String TEST_LOG_DIR = System.getenv().getOrDefault("TEST_LOG_DIR", "/tmp/strimzi");
```
Comment (Frawless) [inline api/src/test/java/io/strimzi/test/StrimziExtension.java]: From my POV the default path to logs should be part of system tests output directory (_./systemtest/target/..._)

File: api/src/test/java/io/strimzi/test/StrimziExtension.java [hunk 2]
```diff
@@ -173,6 +183,61 @@ protected void after() {
...
+            client.pods().list().getItems().forEach(pod -> {
+                String podName = pod.getMetadata().getName();
+
+                client.pods().withName(podName).get().getStatus().getContainerStatuses().forEach(containerStatus -> {
+                    String log = client.pods().withName(podName).inContainer(containerStatus.getName()).getLog();
+                    // Print container logs to console
+                    LOGGER.info("Logs for container {} from pod {}{}{}", containerStatus.getName(), podName, System.lineSeparator(), log);
+
+                    // Write logs from containers to files
+                    writeFile(logDir + "/" + "logs-pod-" + podName + "-container-" + containerStatus.getName() + ".txt", log);
```
Comment (Frawless) [inline api/src/test/java/io/strimzi/test/StrimziExtension.java]: **.log**  extension is better in that case I think then **.txt**

File: api/src/test/java/io/strimzi/test/StrimziExtension.java [hunk 3]
```diff
@@ -173,6 +183,61 @@ protected void after() {
...
+
+        private void collectEvents() {
+            LOGGER.info("Collecting events in namespace {}", namespace);
+            String events = kubeClient().exec("oc", "get", "events").out();
+
+            // Print events to console
+            LOGGER.info("Events for namspace {}{}{}", namespace, System.lineSeparator(), events);
+
+            // Write events to file
+            writeFile(logDir + "/" + "events-in-namespace" + kubeClient().namespace() + ".txt", events);
```
Comment (Frawless) [inline api/src/test/java/io/strimzi/test/StrimziExtension.java]: Same as above

File: api/src/test/java/io/strimzi/test/StrimziExtension.java (hunk 1 above)
Comment (serrss) [reply api/src/test/java/io/strimzi/test/StrimziExtension.java]: Right! This really makes sense. If we will have logs in the target, we will have them in Jenkins's UI.

**Pull request merged**